daemon/HistoryService.xml

```diff
@@ -62,6 +62,27 @@
             <arg type="b" direction="out"/>
             <annotation name="org.qtproject.QtDBus.QtTypeName.In0" value="QList &lt; QVariantMap &gt;"/>
         </method>
+        <method name="RemoveEventsBy">
+            <dox:d><![CDATA[
+                Remove events from the storage given the type and filter
+                Returns true if succeeded in removing the events.
+            ]]></dox:d>
+            <arg name="type" type="i" direction="in"/>
+            <arg name="filter" type="a{sv}" direction="in"/>
+            <arg name="sort" type="a{sv}" direction="in"/>
```

Comment:
it is to avoid having "holes" in the event list. In this implementation, i use the oldest first sorting. If any failure during removing, the newer events will stay

Reply:
Ah, I see, it actually makes sense.